### PR TITLE
Th refactor sleep

### LINKED ISSUE #13: Review sleep and hydration classes
* Review sleep class
* Review hydration class

### LINKED ISSUE #21: Clean up Sleep Testing file using Sample data
* ES6 syntax
* use sample data for testing methods

### PR DESCRIPTION
**Pull request by relyt4me**
### What’s this PR do?
Combing thought the Sleep class and the sleep-test file removed need for userRepo to be pulled into methods. Added sad paths to the test and commented the Sleep file with what methods will replace the ones that have userRepo as a parameter. Also removed the sleepdata from test and built it to user the sampleData

### Where should the reviewer start?
Changes made include the entirety of Sleep.js and sleep-test.js 

### How should this be manually tested?
The 15 test can be run with npm test.

### Any background context you want to provide?
Where the methods are called in scripts will need to be edited to match the new naming convention when that process is ready

### What are the relevant tickets?
resolves #13 
resolves #21 

### Questions:
There are functions at the bottom of Sleep that are not being used but could be rewritten for some decent extenstions.

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -3,77 +3,151 @@ class Sleep {
...
-    let perDaySleep = this.sleepData.filter((data) => id === data.userID);
-    return perDaySleep.reduce((sumSoFar, data) => {
-      return sumSoFar += data.hoursSlept;
-    }, 0) / perDaySleep.length;
+    let sleepForThisUser = this.sleepData.filter((sleep) => id === sleep.userID);
+    const averageSleepHours =
+      sleepForThisUser.reduce((sumSoFar, sleepDay) => {
+        return (sumSoFar += sleepDay.hoursSlept);
+      }, 0) / sleepForThisUser.length;
+    return parseFloat(averageSleepHours.toFixed(2)) || undefined;
```
Comment (carlymclift) [inline src/Sleep.js:11]: Glad you made these methods work for sad paths with the `|| undefined`

File: src/Sleep.js [hunk 2]
```diff
@@ -3,77 +3,151 @@ class Sleep {
...
+  // replaced
   calculateWeekSleepQuality(date, id, userRepo) {
     return userRepo.getWeekFromDate(date, id, this.sleepData).map((data) => `${data.date}: ${data.sleepQuality}`);
   }
+
   calculateAllUserSleepQuality() {
-    var totalSleepQuality = this.sleepData.reduce(function(sumSoFar, dataItem) {
-      sumSoFar += dataItem.sleepQuality;
+    let totalSleepQuality = this.sleepData.reduce((sumSoFar, sleepInstance) => {
+      sumSoFar += sleepInstance.sleepQuality;
```
Comment (carlymclift) [inline src/Sleep.js:65]: Much better naming.

File: src/Sleep.js [hunk 3]
```diff
@@ -3,77 +3,151 @@ class Sleep {
...
+    return uniqueUsers
+      .sort((userA, userB) => {
+        return this.calculateDailySleep(userB, date) - this.calculateDailySleep(userA, date);
+      })
+      .filter((user, index, sortedUniqueUsers) => {
+        return this.calculateDailySleep(user, date) === this.calculateDailySleep(sortedUniqueUsers[0], date);
+      }, []);
   }
+
+  // not sure where all below are being used, cannot find in scripts
```
Comment (carlymclift) [inline src/Sleep.js:123]: All these below can be removed if we don't find them being used?

Comment (relyt4me) [reply src/Sleep.js:123]: Yes, But I think The logic of them could be used if we try for an extension involving other winners.

**Pull request merged**